internal/fsapi/file.go

```diff
@@ -66,4 +70,22 @@ type File interface {
 	//     immediately true, as data will never become available.
 	//   - See /RATIONALE.md for detailed notes including impact of blocking.
 	Poll(flag Pflag, timeoutMillis int32) (ready bool, errno experimentalsys.Errno)
+
+	// OpenAt opens a file with a path relative to this directory.
+	//
+	// # Parameters
+	//
+	// The `path` parameter must be a relative path. The `flag` and `mode`
+	// parameters have the same semantics as [internal/sysfs.OpenFileAt].
+	//
+	// # Notes
+	//
+	//   - This is like `openat` in POSIX.
+	//     See https://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html
+	OpenAt(
+		fs experimentalsys.FS,
+		path string,
+		flag experimentalsys.Oflag,
+		mode fs.FileMode,
+	) (experimentalsys.File, experimentalsys.Errno)
```

Comment:
do we really need to expose this at this level?

the reason I am asking is that this obviously only applies to hierarchical FSs, but it won't work (as you can see later) on other types of file descriptors such as sockets or stdin/err/out. Could it be possible to keep this internal and limited to tree-like file systems?

Essentially, the final goal of this (series of) PR is to introduce an API that will reproduce the correct behavior when resolving paths, so it would great if we can scope it only to files for which "paths" make sense.

Reply:
It definitely doesn't need to be exposed at this level.  I misunderstood your last comment and thought you wanted the method moved from the public interface to a private one.

Reply:
well, that was a fair assumption, but I was wondering if we could keep it as an implementation detail of Open for those FSs that need hierarchical file access 🤔